coalib/settings/FunctionMetadata.py

```diff
@@ -279,12 +285,14 @@ def merge(cls, *metadatas):
             (m.retval_desc for m in reversed(metadatas) if m.retval_desc), "")
         merged_non_optional_params = {}
         merged_optional_params = {}
+        merged_deprecated_params = []
 
         for metadata in metadatas:
             # Use the fields and not the properties to get also omitted
             # parameters.
             merged_non_optional_params.update(metadata._non_optional_params)
             merged_optional_params.update(metadata._optional_params)
+            merged_deprecated_params += metadata.deprecated_params
```

Comment:
`+=` might be undefined if one of them is a list and the other a set, I think update is better